extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java

```diff
@@ -248,11 +250,16 @@ public Response changeUserPassword( ChangePasswordUser user ) {
     return Response.noContent().build();
   }
 
-  private String b64DecodePassword( String encodedPassword ) {
-    if ( !StringUtils.isEmpty( encodedPassword ) && encodedPassword.startsWith( "ENC:" ) ) {
-      return new String( Base64Utils.fromBase64( encodedPassword.substring( 4 ) ), StandardCharsets.UTF_8 );
-    } else {
-      return encodedPassword;
+  private String b64DecodePassword( String encodedPassword ) throws UnsupportedEncodingException {
+    try {
```

Comment:
Please remove the try-catch as it is only rethrowing the original exception (check SonarQube's report).